### PR TITLE
feat: add useResizeObserver option

### PR DESCRIPTION
**Pull request by daybrush**
## Issue

#54

### CODE REVIEW

File: src/ResizeWatcher.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    this.resize();
+    const rect = this.rect;
+    const isWatchWidth = watchDirection === "box" || watchDirection === "width";
+    const isWatchHeight = watchDirection === "box" || watchDirection === "height";
+    const isResize = !watchDirection
+      || (isWatchWidth && prevRect.width !== rect.width)
+      || (isWatchHeight && prevRect.height !== rect.height);
+
+    if (isResize) {
+      this._emitter.trigger("resize", {});
```
Comment (WoodNeck) [inline src/ResizeWatcher.ts]: How about triggering an event alone?
Looks like an empty object is doing nothing in this case.

```ts
private _emitter: Component<{ resize: void }>;

this._emitter.trigger("resize");
```

File: test/unit/Grid.spec.ts [hunk 2]
```diff
@@ -596,6 +626,7 @@ describe("test Grid", () => {
       const spy = sinon.spy();
       grid = new SampleGrid(container!, {
         autoResize: false,
+        resizeDebounce: 0,
```
Comment (WoodNeck) [inline test/unit/Grid.spec.ts]: Why this option is added? `autoResize` is false so I think this isn't needed.
I'm asking this to check whether there isn't any mistake.

File: src/types.ts [hunk 3]
```diff
@@ -30,10 +30,11 @@ import { ItemRenderer, ItemRendererStatus } from "./ItemRenderer";
...
  * @property - The default direction value when direction is not set in the render option. <ko>render옵션에서 direction을 미설정시의 기본 방향값.</ko>
- * @property - You can set the ItemRenderer directly externally. <ko>외부에서 직접 ItemRenderer를 설정할 수 있다.</ko>
- * @property - You can set the ContainerManager를 directly externally. <ko>외부에서 직접 ContainerManager를 설정할 수 있다.</ko>
  * @property - Whether to get the size as rounded size(offsetWidth, offsetHeight). Set to true if transform is applied to the container. If false, get the size through getBoundingClientRect. (default: true)
  <ko>사이즈를 반올림된 사이즈(offsetWidth, offsetHeight)로 가져올지 여부. container에 transform이 적용되어 있다면 true로 설정해라. false면 getBoundingClientRect를 통해 사이즈를 가져온다. (default: true)</ko>
+ * @property - Whether to use ResizeObserver event to detect container size change when `autoResize` option is used.<ko>autoResize 옵션 사용시 container의 사이즈 변화 감지를 위해 ResizeObserver 이벤트를 사용할지 여부.</ko>
+ * @property - You can set the ItemRenderer directly externally. <ko>외부에서 직접 ItemRenderer를 설정할 수 있다.</ko>
+ * @property - You can set the ContainerManager를 directly externally. <ko>외부에서 직접 ContainerManager를 설정할 수 있다.</ko>
```
Comment (WoodNeck) [inline src/types.ts]: Typo in the description, mix of English - Korean

File: package.json [hunk 4]
```diff
@@ -146,7 +146,7 @@
 		"storybook-dark-mode": "^1.0.3",
 		"ts-node": "^9.1.1",
 		"tslib": "^2.0.3",
-		"typescript": "^3.9.7"
+		"typescript": "^4.2.4"
```
Comment (WoodNeck) [inline package.json:149]: Is this version update needed?

File: src/ResizeWatcher.ts [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+
+export class ResizeWatcher {
+  private _resizeTimer = 0;
+  private _maxResizeDebounceTimer = 0;
+  private _emitter: Component<{ resize: OnResize }>;
+  private _observer: ResizeObserver | null;
+  protected container: HTMLElement;
+  protected rect: SizeRect = { width: 0, height: 0 };
+  protected options!: Required<ResizeWatherOptions>;
```
Comment (WoodNeck) [inline src/ResizeWatcher.ts]: Why these properties are marked as `protected`? Is there any class that inherits this class?

**Review by WoodNeck — APPROVED** [review]
LGTM

File: package.json (hunk 4 above)
Comment (daybrush) [reply package.json:149]: @WoodNeck 

ResizeObserver is supported from typescript version 4 or later.

**Review by malangfox — APPROVED** [review]
LGTM

**Pull request merged**